File: my-games-backend/src/main/java/com/tc/mygames/dto/GameDTO.java
package com.tc.mygames.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import javax.validation.constraints.*;
import java.time.LocalDate;

@Data
@AllArgsConstructor
public class GameDTO {

    @NotEmpty(message = "error.invalid.title.not.empty")
    private String title;

    @PastOrPresent(message = "error.invalid.completionDate.not.future")
    private LocalDate completionDate;

    @NotEmpty(message = "error.invalid.console.not.empty")
    private String console;

    @NotNull(message = "error.invalid.year.not.null")
    private LocalDate year;

    private String notes;

    @NotNull(message = "error.invalid.completed.not.null")
    private Boolean completed;
}

File: my-games-frontend/src/model/GameDTO.ts
class Game {
  public id: String;

  public title: String;

  public completionDate: string;

  public console: String;

  public year: string;

  public notes: String;

  public completed: Boolean;

  constructor(
    id: String,
    title: String,
    completionDate: string,
    console: String,
    year: string,
    notes: String,
    completed: Boolean,
  ) {
    this.id = id;
    this.title = title;
    this.completionDate = completionDate;
    this.console = console;
    this.year = year;
    this.notes = notes;
    this.completed = completed;
  }
}

export default Game;
